### PR TITLE
feat(web-client):  add country code & validation to register page

### PR DESCRIPTION
**Pull request by jongbonga**
Allow to format phone number to have prefix

### CODE REVIEW

**Comment by PiDelport** [comment]
ion-intl-tel-input is probably a better approach here, if we can get the styling fixed?

- https://github.com/ntls-io/nautilus-wallet/pull/231

Regardless of UI choice, the crucial thing is that we need the numbers normalised to E.164, so we'll almost certainly want to use libphonenumber; see <https://github.com/ntls-io/nautilus-wallet/issues/215#issuecomment-1122294364>.

**Comment by jongbonga** [comment]
> ion-intl-tel-input is probably a better approach here, if we can get the styling fixed? See this PR:
> 
> * [feat(web-client): use ion-intl-tel-input for mobile number input #231](https://github.com/ntls-io/nautilus-wallet/pull/231)
> 
> Regardless of UI choice, the crucial thing is that we need the numbers normalised to E.164, so we'll almost certainly want to use libphonenumber; see [#215 (comment)](https://github.com/ntls-io/nautilus-wallet/issues/215#issuecomment-1122294364).

updated to use the library

File: web-client/src/app/app.module.ts [hunk 1]
```diff
@@ -27,6 +28,7 @@ import { TranslocoRootModule } from './transloco/transloco-root.module';
     AkitaNgRouterStoreModule,
     NgxPrinterModule.forRoot({ printOpenWindow: false }),
     ErrorHandlerModule,
+    IonIntlTelInputModule,
```
Comment (longtomjr) [inline web-client/src/app/app.module.ts:31]: Do we need to import this in the app.module and the register.module?

**Review by longtomjr — APPROVED** [review]
Looks good.

Maybe just change this section in the session service as well, since I don't think it is needed anymore.
https://github.com/ntls-io/nautilus-wallet/blob/68c5358cbde2cdacb05abfb5786237fcf78c8ae7/web-client/src/app/state/session.service.ts#L164-L167

File: web-client/src/app/app.module.ts (hunk 1 above)
Comment (jongbonga) [reply web-client/src/app/app.module.ts:31]: No, thanks!

**Pull request merged**